examples/cosmos/ics20/bank.qnt

```diff
@@ -0,0 +1,240 @@
+   */
+  pure def manipulateAccount(accounts: Accounts, address: Address,
+                             denom: Denomination, amount: Amount) : Accounts = {
+    val accountCoins = accounts.getBalances(address)
+    val newCoins = accountCoins.put(denom, accountCoins.getBalance(denom) + amount)
+    accounts.put(address, newCoins)
+  }
+
+  /* Return `accounts`, where `amount`-many tokens of denomination `denom`
+   * vanished out of `sender`'s account into thin air.
+   *
+   * This intentionally does NOT check whether `amount > 0`.
+   */
+  pure def BurnCoins(accounts: Accounts, sender: Address,
+                     denom: Denomination, amount: Amount): Accounts = {
+    manipulateAccount(accounts, sender, denom, -amount)
+  }
+
+  /* Return `accounts`, where `amount`-many tokens of denomination `denom` were
+   * added to `sender`'s account out of thin air.
+   *
+   * This intentionally does NOT check whether `amount > 0`.
+   */
+  pure def MintCoins(accounts: Accounts, receiver: Address,
+                     denom: Denomination, amount: Amount): Accounts = {
+    manipulateAccount(accounts, receiver, denom, amount)
+  }
+
+  /* Return `state`, where `amount`-many tokens of denomination `denom` were
+   * moved from `sender`'s account into `receiver`'s acount.
+   *
+   * This intentionally does NOT check whether `amount > 0`.
+   */
+  pure def TransferCoins(accounts: Accounts, sender: Address, receiver: Address,
+                         denom: Denomination, amount: Amount): Accounts = {
+    accounts
+      .BurnCoins(sender, denom, amount)
+      .MintCoins(receiver, denom, amount)
+  }
+
+  /* ***************************************************************************
+   * STATE MACHINE
+   * **************************************************************************/
+
+  var State: Accounts
+  pure val Addresses : Set[Address] = Set("alice", "bob", "charlie")
+  pure val Denoms : Set[Denomination] = Set("atom", "eth", "osmo")
+
+  // Trivial action defs for the bank module operations
+  action burn(sender: Address, denom: Denomination, amount: Amount): bool =
+    State' = State.BurnCoins(sender, denom, amount)
+
+  action mint(receiver: Address, denom: Denomination, amount: Amount): bool =
+    State' = State.MintCoins(receiver, denom, amount)
+
+  action transfer(sender: Address, receiver: Address,
+                  denom: Denomination, amount: Amount): bool =
+    State' = State.TransferCoins(sender, receiver, denom, amount)
+
+  /* Initialize accounts.
+   *
+   * Alice, Bob and Charlie initially hold 10 ATOMs, and no other coins.
+   */
+  action init = State' = Addresses.mapBy(_ => Map("atom" -> 10))
+
+  /* Nondeterministically burn, mint, or transfer tokens of arbitrary
+   * denomination and amount to/from arbitrary accounts.
+   */
+  action step = {
+    nondet addrA = Addresses.oneOf()
+    nondet addrB = Addresses.oneOf()
+    nondet denom = Denoms.oneOf()
+    nondet amount = Int.oneOf()
```

Comment:
```suggestion
    // since there are no assumptions about the range of amounts, this can be an arbitrary integer!
    nondet amount = Int.oneOf()
```